pairinteraction_gui/pairinteraction/app.py

```diff
@@ -483,21 +483,34 @@ def __init__(self, parent=None):
         self.path_cpp_complex = os.path.join(self.path_base, self.path_workingdir, "pairinteraction-complex")
         self.path_quantumdefects = os.path.join(self.path_base, self.path_workingdir, "databases/quantum_defects.db")
 
-        if os.name == "nt":
-            self.path_out = os.path.join(self.userpath, "pairinteraction/")
+        if sys.platform == "win32":
+            self.path_config = os.path.expandvars(r"%APPDATA%\pairinteraction")
+            if r"%APPDATA%" in self.path_config:
+                self.path_config = os.path.expanduser(r"~\AppData\Local\pairinteraction")
+            self.path_cache = os.path.expandvars(r"%LOCALAPPDATA%\pairinteraction")
+            if r"%LOCALAPPDATA%" in self.path_config:
+                self.path_cache = os.path.expanduser(r"~\AppData\Local\pairinteraction")
+        elif sys.platform == "darwin":
+            self.path_config = os.path.expanduser(r"~/Library/Preferences/pairinteraction")
+            self.path_cache = os.path.expanduser(r"~/Library/Caches/pairinteraction")
         else:
-            self.path_out = os.path.join(self.userpath, ".pairinteraction/")
-        self.path_cache = os.path.join(self.path_out, "cache/")
+            self.path_config = os.path.expandvars(r"$XDG_CONFIG_HOME/pairinteraction")
+            if r"$XDG_CONFIG_HOME" in self.path_config:
+                self.path_config = os.path.expanduser(r"~/.config/pairinteraction")
+            self.path_cache = os.path.expandvars(r"$XDG_CACHE_HOME/pairinteraction")
+            if r"$XDG_CACHE_HOME" in self.path_cache:
+                self.path_cache = os.path.expanduser(r"~/.cache/pairinteraction")
+
         self.path_cache_wignerd = os.path.join(self.path_cache, "wignerd/")
-        self.path_lastsettings = os.path.join(self.path_out, "lastsettings/")
+        self.path_lastsettings = os.path.join(self.path_config, "lastsettings/")
 
         self.path_system_last = os.path.join(self.path_lastsettings, "lastsettings.sconf")
         self.path_plot_last = os.path.join(self.path_lastsettings, "lastsettings.pconf")
         self.path_view_last = os.path.join(self.path_lastsettings, "lastview.json")
         self.path_cache_last = os.path.join(self.path_lastsettings, "lastcache.json")
 
-        self.path_config = os.path.join(self.path_out, "conf.json")
-        self.path_version = os.path.join(self.path_out, "version.json")
+        self.path_config = os.path.join(self.path_config, "conf.json")
+        self.path_version = os.path.join(self.path_config, "version.json")
```

Comment:
These lines set `self.path_version` to "[...]/conf.json/version.json", but it should be "[...]/version.json". To fix this bug, you could rename `self.path_out` to `self.path_preferences` instead of renaming it to `self.path_config`.